src/Core/src/AbstractApi.php

```diff
@@ -113,11 +114,17 @@ public function __construct($configuration = [], ?CredentialProvider $credential
         $request = new Request($method, $this->fillEndpoint($endpoint), $headers, $stream);
         $this->getSigner()->sign($request, $this->credentialProvider->getCredentials($this->configuration));
 
-        if (!$request->hasHeader('content-length') && null !== $length = $request->getBody()->length()) {
+        if (null !== ($length = $request->getBody()->length()) && !$request->hasHeader('content-length')) {
```

Comment:
Could you please write this in a more readable way?

Reply:
done. Thanks